src/IECoreScene/ShaderNetworkAlgo.cpp

```diff
@@ -684,6 +713,213 @@ IECore::DataPtr loadSpline(
+						{
+							adapter->parameters()[ g_arrayInputNames[i] ] = new FloatData( splineValues[i] );
+						}
+					}
+				}
+
+				if( adapterArraySize > maxArrayInputAdapterSize )
+				{
+					throw IECore::Exception(
+						"Cannot handle input to " + connection.destination.shader.string() + "." +
+						destName + ". Expanded spline has " + std::to_string( adapterArraySize) +
+						" control points, but max input adapter size is " + std::to_string( maxArrayInputAdapterSize )
+					);
+				}
+
+				// TODO - does this need to be tagged to be reversible?
+				// ANSWER - it's a special node type, which users shouldn't be creating, so John pointed out
+				// we can probably just remove them all without needing to tag
+				//adapter->blindData()->writable()[ componentConnectionAdapterLabel() ] = g_trueData;
+
+				adapterHandle = network->addShader( s.first.string() + "InputArrayAdapter", std::move( adapter ) );
+				network->addConnection( ShaderNetwork::Connection(
+					{ adapterHandle, g_arrayOutputNames[ adapterArraySize ] },
+					{ s.first, splineValuesName }
+				) );
+
+				inserted.first->second.first = adapterHandle;
+				inserted.first->second.second = knotOffset;
+			}
+			else
+			{
+				adapterHandle = inserted.first->second.first;
+				knotOffset = inserted.first->second.second;
+			}
+
+			// Default is guaranteed not to be used, since from_chars must either set it or set the error code,
+			// but apparently the compiler doesn't know that
+			int elementId = -1;
+			const char *idStart = &destName[splineElementMatch.position(2)];
+			const char *idEnd = idStart + splineElementMatch.length(2);
+			auto elementIdResult = std::from_chars( idStart, idEnd, elementId );
+
+			if( elementIdResult.ec == std::errc::invalid_argument || elementIdResult.ptr != idEnd )
+			{
+				throw IECore::Exception( "Invalid spline point index " + splineElementMatch[2] );
+			}
+
+			InternedString destinationName = g_arrayInputNames[elementId + knotOffset];
+			if( splineElementMatch.length( 3 ) )
+			{
+				std::string nameWithSuffix;
+				nameWithSuffix.reserve( destinationName.string().size() + splineElementMatch.length(3) );
+				nameWithSuffix.append( destinationName.string() );
+				nameWithSuffix.append( &destName[splineElementMatch.position(3)], splineElementMatch.length(3) );
+
+				destinationName = nameWithSuffix;
+			}
+
+			network->removeConnection( connection );
+			network->addConnection( { connection.source, { adapterHandle, destinationName } } );
+		}
+
+		network->setShader( s.first, std::move( new Shader( s.second->getName(), s.second->getType(), newParametersData.get() ) ) );
+
+	}
+
+	// \todo - it would be a bit more efficient to integrate this, and only traverse the network once,
+	// but I don't think it's worth duplicated the code - fix this up once this call is standard and we
+	// deprecate and remove convertOSLComponentConnections
+	convertOSLComponentConnections( network, oslVersion);
```

Comment:
Two thoughts :

1. I think an all-in-one function might need to be structured such that it was responsible for the outer loop and then farmed the work out to various `convertFoo( shader, newParameters, newConnections  )` functions. That might allow `convertOSLComponentConnections()` to be implemented similarly, with very little duplication. I think the added structure might also help readability anyway.
2. We could hold this feature till Gaffer 1.3 and Cortex 10.5, giving us carte-blanche to throw out the stuff we don't want.